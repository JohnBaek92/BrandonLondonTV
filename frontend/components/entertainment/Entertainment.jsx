import React from "react";
import Youtube from "react-youtube";


class Entertainment extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      openModal: false,
      picture: null,
      prevVideo: null,
      top: "50vh",
      middle: "25vh",
      bottom: "15vh",
      videos: {
        E8JlWMXJgqA: false,
        "BKXLB4-9sfg": false,
        KLmwpPbTsN0: false,
        rJ_qxnuKADY: false,
        YbXrdVjzkC4: false,
        AOmHwHWpdKc: false,
        erRJtA8bkVc: false,
        "WSmNvV-8uwU": false
      }
    };

    this.videos = ["Dxs596KF7-Y", "NXLMJLd3Wbw", "Bqz9m6dhA-M"];
    this.ontheroad = ["KLmwpPbTsN0", "4lRkQZTapcg", "erRJtA8bkVc", "J_MAIrPsmRA"];
    this.bestmoves = ["unxtRCOsZpE", "z7unu1gYNwU", "BKXLB4-9sfg", "pfL0FbCBomw", "Y9YP2EQJ_wI", "AOmHwHWpdKc"];
    this.fanexperience = ["7WOW3hCIg8U", "G4iZJtXw5ig", "rJ_qxnuKADY", "y1_sZeSDlc8"];
    this.asSeenOnTV = ["Dxs596KF7-Y", "NXLMJLd3Wbw", "Bqz9m6dhA-M"];
    this.goplay = ["WSmNvV-8uwU"];
    this.changeHeight = this.changeHeight.bind(this);
    this.makeVideo = this.makeVideo.bind(this);
    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  changeHeight(row, idx, link, e) {
    const newState = Object.assign({}, this.state);
    if (this.state.prevVideo) {
      this.state.prevVideo.pauseVideo();
    }
    newState.top = "15vh";
    newState.middle = "15vh";
    newState.bottom = "15vh";
    newState[row] = "50vh";
    newState.videos = {
      E8JlWMXJgqA: false,
      "BKXLB4-9sfg": false,
      KLmwpPbTsN0: false,
      rJ_qxnuKADY: false,
      YbXrdVjzkC4: false,
      AOmHwHWpdKc: false,
      erRJtA8bkVc: false,
      "WSmNvV-8uwU": false
    };
    newState.videos[link] = true;
    newState.prevVideo = e.target;
    this.setState(newState);
  }

  makeVideo(link, row, idx) {
    let active = "column";
    let videoSrc = this.state.videos[link];
    if (videoSrc) {
      active = "column is-three-quarters";
    }
    return (
      <div className={active} key={link}>
        <Youtube
          videoId={link}
          opts={{
            height: "100%",
            width: "100%"
          }}
          // onPlay={e => this.changeHeight(row, idx, link, e)}
        />
      </div>
    );
  }

  openModal() {
    return (
      <div className="modal-body" onClick={this.closeModal}>
        <div className="modal-content">
          <img src={this.state.picture} />
        </div>
      </div>
    );
  }

  openModalState(e) {
    if (!e.target.src) {
      return null;
    }
    this.setState({ openModal: true, picture: e.target.src });
  }

  closeModal() {
    this.setState({ openModal: false });
  }

  componentWillMount() {
    window.scrollTo(0, 0);
  }

  render() {
    return <div className="entertainment-wrapper" onClick={e => this.openModalState(e)}>
        {this.state.openModal ? this.openModal() : null}
        <div className="title has-text-centered">Photos</div>
        <div className="columns is-mobile">
          <img className="column" src={window.images.e1} />
          <img className="column" src={window.images.e5} />
          <img className="column" src={window.images.e3} />
          <img className="column" src={window.images.e4} />
        </div>
        <div className="columns is-mobile">
          <img className="column" src={window.images.e2} />
          <img className="column" src={window.images.e6} />
          <img className="column" src={window.images.e7} />
          <img className="column" src={window.images.e8} />
        </div>
        <div className="title has-text-centered">Videos</div>
        <div className="subtitle has-text-centered">My Best Moves - Player Interviews</div>
        <div className="columns is-mobile" style={{ height: this.state.top }}>
          {this.bestmoves
            .slice(0, 2)
            .map(link => this.makeVideo(link, "middle"))}
        </div>
        <div className="columns is-mobile" style={{ height: this.state.top }}>
          {this.bestmoves
            .slice(2, 4)
            .map(link => this.makeVideo(link, "middle"))}
        </div>
        <div className="columns is-mobile" style={{ height: this.state.top }}>
          {this.bestmoves
            .slice(4, 6)
            .map(link => this.makeVideo(link, "middle"))}
        </div>
        <div className="subtitle has-text-centered">Fan Experience</div>
        <div className="columns is-mobile" style={{ height: this.state.top }}>
          {this.fanexperience
            .slice(0, 2)
            .map(link => this.makeVideo(link, "middle"))}
        </div>
        <div className="columns is-mobile" style={{ height: this.state.top }}>
          {this.fanexperience
            .slice(2, 4)
            .map((link, idx) => this.makeVideo(link, "middle", idx))}
        </div>
        <div className="subtitle has-text-centered">On The Road</div>
        <div className="columns is-mobile" style={{ height: this.state.top }}>
          {this.ontheroad
            .slice(0, 2)
            .map((link, idx) => this.makeVideo(link, "middle", idx))}
        </div>
        <div className="columns is-mobile" style={{ height: this.state.top }}>
          {this.ontheroad
            .slice(2, 4)
            .map((link, idx) => this.makeVideo(link, "middle", idx))}
        </div>
        <div className="subtitle has-text-centered">Giants Play 60</div>
        <div className="columns is-mobile" style={{ height: this.state.top }}>
          {this.goplay
            .slice(0, 2)
            .map((link, idx) => this.makeVideo(link, "middle", idx))}
        </div>
        <div className="subtitle has-text-centered">As Seen On TV</div>
        <div className="columns is-mobile" style={{ height: this.state.top }}>
          {this.asSeenOnTV
            .slice(0, 2)
            .map((link, idx) => this.makeVideo(link, "middle", idx))}
        </div>
        <div className="columns is-mobile" style={{ height: this.state.top }}>
          {this.asSeenOnTV
            .slice(2, 4)
            .map((link, idx) => this.makeVideo(link, "middle", idx))}
        </div>
      </div>;
  }
}

export default Entertainment;
