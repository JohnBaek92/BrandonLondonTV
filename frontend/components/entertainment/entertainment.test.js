import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Entertainment from "./Entertainment";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

test("Entertainment component should render as expected", () => {
  const component = shallow(<Entertainment />);
  const tree = toJson(component);

  expect(tree).toMatchSnapshot();
});
