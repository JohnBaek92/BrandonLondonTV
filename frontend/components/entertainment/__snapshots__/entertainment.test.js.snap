// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entertainment component should render as expected Renders the Entertainment correctly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "2jd09b",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "2jd09b",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Entertainment>
      <section
        className="entertainment-wrapper"
      >
        <div
          className="title has-text-centered"
        >
          Photos
        </div>
        <div
          className="columns is-mobile"
        >
          <img
            className="column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "height": "15%",
                "width": "15%",
              }
            }
          />
          <img
            className="column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "height": "15%",
                "width": "15%",
              }
            }
          />
          <img
            className="column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "height": "15%",
                "width": "15%",
              }
            }
          />
          <img
            className="column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "height": "15%",
                "width": "15%",
              }
            }
          />
        </div>
        <div
          className="columns is-mobile"
        >
          <img
            className="column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "height": "15%",
                "width": "15%",
              }
            }
          />
          <img
            className="column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "height": "15%",
                "width": "15%",
              }
            }
          />
          <img
            className="column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "height": "15%",
                "width": "15%",
              }
            }
          />
          <img
            className="column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "height": "15%",
                "width": "15%",
              }
            }
          />
        </div>
        <div
          className="title has-text-centered"
        >
          Videos
        </div>
        <div
          className="columns is-mobile"
          style={
            Object {
              "height": "35vh",
            }
          }
        >
          <div
            className="column"
            key="E8JlWMXJgqA"
          >
            <YouTube
              onEnd={[Function]}
              onError={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onPlaybackQualityChange={[Function]}
              onPlaybackRateChange={[Function]}
              onReady={[Function]}
              onStateChange={[Function]}
              opts={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
              videoId="E8JlWMXJgqA"
            >
              <span>
                <div />
              </span>
            </YouTube>
          </div>
          <div
            className="column"
            key="BKXLB4-9sfg"
          >
            <YouTube
              onEnd={[Function]}
              onError={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onPlaybackQualityChange={[Function]}
              onPlaybackRateChange={[Function]}
              onReady={[Function]}
              onStateChange={[Function]}
              opts={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
              videoId="BKXLB4-9sfg"
            >
              <span>
                <div />
              </span>
            </YouTube>
          </div>
        </div>
        <div
          className="columns is-mobile"
          style={
            Object {
              "height": "25vh",
            }
          }
        >
          <div
            className="column"
            key="KLmwpPbTsN0"
          >
            <YouTube
              onEnd={[Function]}
              onError={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onPlaybackQualityChange={[Function]}
              onPlaybackRateChange={[Function]}
              onReady={[Function]}
              onStateChange={[Function]}
              opts={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
              videoId="KLmwpPbTsN0"
            >
              <span>
                <div />
              </span>
            </YouTube>
          </div>
          <div
            className="column"
            key="rJ_qxnuKADY"
          >
            <YouTube
              onEnd={[Function]}
              onError={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onPlaybackQualityChange={[Function]}
              onPlaybackRateChange={[Function]}
              onReady={[Function]}
              onStateChange={[Function]}
              opts={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
              videoId="rJ_qxnuKADY"
            >
              <span>
                <div />
              </span>
            </YouTube>
          </div>
        </div>
        <div
          className="columns is-mobile"
          style={
            Object {
              "height": "15vh",
            }
          }
        >
          <div
            className="column"
            key="YbXrdVjzkC4"
          >
            <YouTube
              onEnd={[Function]}
              onError={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onPlaybackQualityChange={[Function]}
              onPlaybackRateChange={[Function]}
              onReady={[Function]}
              onStateChange={[Function]}
              opts={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
              videoId="YbXrdVjzkC4"
            >
              <span>
                <div />
              </span>
            </YouTube>
          </div>
          <div
            className="column"
            key="AOmHwHWpdKc"
          >
            <YouTube
              onEnd={[Function]}
              onError={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onPlaybackQualityChange={[Function]}
              onPlaybackRateChange={[Function]}
              onReady={[Function]}
              onStateChange={[Function]}
              opts={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
              videoId="AOmHwHWpdKc"
            >
              <span>
                <div />
              </span>
            </YouTube>
          </div>
          <div
            className="column"
            key="erRJtA8bkVc"
          >
            <YouTube
              onEnd={[Function]}
              onError={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onPlaybackQualityChange={[Function]}
              onPlaybackRateChange={[Function]}
              onReady={[Function]}
              onStateChange={[Function]}
              opts={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
              videoId="erRJtA8bkVc"
            >
              <span>
                <div />
              </span>
            </YouTube>
          </div>
          <div
            className="column"
            key="WSmNvV-8uwU"
          >
            <YouTube
              onEnd={[Function]}
              onError={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onPlaybackQualityChange={[Function]}
              onPlaybackRateChange={[Function]}
              onReady={[Function]}
              onStateChange={[Function]}
              opts={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
              videoId="WSmNvV-8uwU"
            >
              <span>
                <div />
              </span>
            </YouTube>
          </div>
        </div>
      </section>
    </Entertainment>
  </Router>
</MemoryRouter>
`;
